src/MiniProfiler.Shared/Timing.cs

```diff
@@ -77,28 +78,76 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public List<Timing> Children
+        {
+            get
+            {
+                lock(_lockObject)
+                {
+                    return _children == null ? null : new List<Timing>(_children);
+                }
+            }
+            set
+            {
+                lock(_lockObject)
+                {
+                    _children = value == null ? null : new List<Timing>(value);
+                }
+            }
+        }
 
         /// <summary>
         /// <see cref="CustomTiming"/> lists keyed by their type, e.g. "sql", "memcache", "redis", "http".
         /// </summary>
         [DataMember(Order = 6)]
-        public Dictionary<string, List<CustomTiming>> CustomTimings { get; set; }
+        public Dictionary<string, List<CustomTiming>> CustomTimings
```

Comment:
These will add overhead. As an example we run this for *every single request* at Stack Overflow. So the profiler can't be in the way or causing contention. As an example, we have mobile feed threads that may returns tens of thousands of tasks in a very narrow window. - Some of them will wait on the locks.

Equally important, anyone accessing these gets a copies. This really is a no-go under any circumstances. One of the founding principles of **Mini**Profiler is the lightweight nature. The amount of allocations involved for operational overhead here is simply untenable. We'll have to find another approach.